reana_workflow_controller/tasks.py

```diff
@@ -99,3 +107,42 @@ def _update_job_progress(workflow_uuid, msg):
                     run_job.run_id = current_run.id_
                     run_job.job_id = job_id
                     Session.add(run_job)
+
+
+def _update_job_cache(msg):
+    """Update caching information for finished job."""
+    cached_job = Session.query(JobCache).filter_by(
+        job_id=msg['caching_info'].get('job_id')).first()
+
+    input_files = []
+    if cached_job:
+        file_access_times = calculate_file_access_time(
+            msg['caching_info'].
+            get('workflow_workspace').
+            replace('data', 'reana/default'))
+        for filename in cached_job.access_times:
+            if filename in file_access_times and \
+                    file_access_times[filename] > \
+                    cached_job.access_times[filename]:
+                input_files.append(filename)
+    else:
+        return
+    cmd = msg['caching_info']['job_spec']['cmd']
+    # removes cd to workspace, to be refactored
+    clean_cmd = cmd.split(';')[1]
```

Comment:
shall we `';'.join(test.split(';')[1:])` since most probably there will be more `;` inside the command?

Reply:
good point, fixed here and in `r-w-e-serial` 👍